src/mlm_server.c

```diff
@@ -703,8 +703,10 @@ deregister_the_client (client_t *self)
             }
             service = (service_t *) zhashx_next (self->server->services);
         }
-        if (*self->address)
+        if (*self->address) {
             zhashx_delete (self->server->clients, self->address);
+            zstr_free (&self->address);
```

Comment:
Yes, sure. I'll make another PR for this. But it's just weird to have a destructor that doesn't actually destroy the instance, just some of its internal properties.

client_terminate () should be the routine responsible for this I think, right?